app/2.0.x/go.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Go language support
+---
+
+## Introduction
+
+Kong 2.0 introduces support for plugins written in Go.
+
+## Prerequisites
+
+To use Go plugins:
+
+* You need the `go-pluginserver` executable.  The Kong distribution packages
+install it in `/usr/local/bin/`.  If you want it anywhere else, set the
+`go_pluginserver_exe` variable in the Kong configuration file to the full path.
+* Set the `go_plugins_dir` variable in the Kong configuration file to a valid directory.
+The default `"off"` value disables Go support.
+* Add compiled Go plugins to the directory specified in the previous step.
+* Add the plugin settings the usual way, either via Admin API, database or declarative file.
+Refer to the the plugin via it's filename (without the `.so` suffix).
+
+To write your own Go plugins:
+
+* A Kong development environment as usual.
+* A Go development environment.
+* The Go-PDK package: `go get github.com/Kong/go-pdk`
+
+## Development
+
+To write a Kong plugin in Go, you need to:
+
+* Define a structure type to hold configuration.
+* Write a `New()` function to create instances of your structure.
+* Add methods on that structure to handle events
+* Compile like: `go build -buildmode=plugin`.
+* Put the resulting library (the `.so` file) into the `go_plugins_dir` directory.
+
+Go support for binary plugins requires very tight matching of compiler and library versions
+between the executable and plugin.  We've used Go compiler version 1.13.5 although this might
+change frequently.
+If you get startup errors complaining about version
+mismatch, you can recompile the `go-pluginserver` executable:
+
+```bash
+# git clone https://github.com/Kong/go-pluginserver
+# cd go-pluginserver
+# make
+```
+
+Point the `go_pluginserver_exe` configuration variable to your new version.
+
```

Comment:
Could we link to https://github.com/Kong/go-plugins somewhere? It might be useful as a reference.